shenyu-admin/src/test/java/org/apache/shenyu/admin/service/UpstreamCheckServiceTest.java

```diff
@@ -66,15 +90,53 @@
 
     private Map<String, List<DivideUpstream>> upstreamMap;
 
+    private Set<ZombieUpstream> zombieSet;
+
     @Before
     public void setUp() {
         shenyuRegisterCenterConfig.setRegisterType("http");
+        Properties properties = new Properties();
+        properties.setProperty(Constants.IS_CHECKED, "true");
+        shenyuRegisterCenterConfig.setProps(properties);
 
         //get static variable reference by reflection
         upstreamMap = (Map<String, List<DivideUpstream>>) ReflectionTestUtils.getField(UpstreamCheckService.class, "UPSTREAM_MAP");
+        zombieSet = (Set<ZombieUpstream>) ReflectionTestUtils.getField(UpstreamCheckService.class, "ZOMBIE_SET");
+
+        //mock data
+        PluginDO pluginDO = PluginDO.builder()
+                .name(PluginEnum.DIVIDE.getName())
+                .id(MOCK_PLUGIN_ID)
+                .build();
+        SelectorDO selectorDOWithUrlError = SelectorDO.builder()
+                .pluginId(MOCK_PLUGIN_ID)
+                .name(MOCK_SELECTOR_NAME)
+                .handle("[{\"upstreamHost\":\"localhost\",\"protocol\":\"http://\",\"upstreamUrl\":\"divide-upstream-50\",\"weight\":50}]")
+                .build();
+        SelectorDO selectorDOWithUrlReachable = SelectorDO.builder()
+                .pluginId(MOCK_PLUGIN_ID)
+                .name(MOCK_SELECTOR_NAME_OTHER)
+                .handle("[{\"upstreamHost\":\"localhost\",\"protocol\":\"http://\",\"localhost\":\"divide-upstream-60\",\"weight\":60}]")
+                .build();
+
+        //stubbing
+        when(pluginMapper.selectByNames(anyList())).thenReturn(Lists.newArrayList(pluginDO));
+        when(pluginMapper.selectById(anyString())).thenReturn(pluginDO);
+        when(selectorMapper.findByPluginId(anyString())).thenReturn(Lists.newArrayList(selectorDOWithUrlError, selectorDOWithUrlReachable));
+        when(selectorMapper.updateSelective(any(SelectorDO.class))).thenReturn(1);
+        when(selectorMapper.selectByName(anyString())).then(invocationOnMock -> {
+            Object[] args = invocationOnMock.getArguments();
+            if (MOCK_SELECTOR_NAME.equals(args[0])) {
+                return selectorDOWithUrlError;
+            } else if (MOCK_SELECTOR_NAME_OTHER.equals(args[0])) {
+                return selectorDOWithUrlReachable;
+            }
+            return null;
+        });
+        when(selectorConditionMapper.selectByQuery(any(SelectorConditionQuery.class))).thenReturn(Collections.emptyList());
+        doNothing().when(eventPublisher).publishEvent(any(DataChangedEvent.class));
```

Comment:
These codes already removed by [https://github.com/apache/incubator-shenyu/pull/1843](https://github.com/apache/incubator-shenyu/pull/1843). Please check again. Thanks.

Reply:
@dengliming Thanks for reminding. Yep, I check it, there are some unnecessary stubbings, and I delete it. But, some should be left, or the coverage will drop.

Reply:
I mean we should remove all unnecessary stubbings.

Reply:
@dengliming ok, I have refactored it, please have a review again. There should be no unnecessary stubbings in UpstreamCheckServiceTest.